src/components/Table/components/Row.tsx

```diff
@@ -320,22 +309,17 @@ function RowImpl<R extends Kinded, S>(props: RowProps<R>): ReactElement {
 }
 
 /**
- * Memoizes Rows so that re-rendering the table doesn't re-render every single row.
+ * Memoizes rows so that re-rendering the table doesn't re-render every single row.
  *
- * We use a custom `propsAreEqual` for the `RowProps.row` property, which we memoize
- * based on the `GridDataRow.data` prop, such that if a table re-renders (i.e. for a cache
- * updated) and technically creates new row instances, but a row's `row.data` points to the
- * same/unchanged Apollo cache fragment, then we won't re-render that row.
+ * We used to have a custom `propsAreEqual` method that would only compare `props.rows`
+ * on the `data` attribute, b/c our `createRows` methods do not create stable row identities;
+ * i.e. if one row changes, they all change.
  *
- * Note that if you're using virtualization, it can be surprising how unnoticeable broken row
- * memoization is.
+ * However, now RowState.row synthesizes a `row` that only reactively changes when
+ * that row's `data` explicitly changes, so we no longer need a custom `propsAreEqual` here.
  */
 // Declared as a const + `as typeof RowImpl` to work with generics, see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/37087#issuecomment-656596623
-export const Row = React.memo(observer(RowImpl), (one, two) => {
```

Comment:
Also turns out that `observer` internally does a `React.memo`, so we don't need to include one ourselves.